[MLIR][OpenMP]Add prescriptiveness-modifier support to grainsize and …

kaviya2510 opened the pull request:
…num_tasks clause.

kaviya2510 committed:
[MLIR][OpenMP]Add prescriptiveness-modifier support to grainsize and num_tasks clause.

tblah reviewed (commented):
These two parsing functions are doing broadly the same thing. Could they share code?

kaviya2510 committed:
[MLIR][OpenMP]Refactored parser and printer function of grainsize and numtasks clause.

kaviya2510 commented:
Thanks for the comments @tblah .
Yes, we can share the code for parser and printer functions for grainsize and numtasks clause. I have modified and updated the changes, Kindly take a look it.
Thankyou.

tblah reviewed (approved):
LGTM, thanks for the update

skatrak commented on mlir/lib/Dialect/OpenMP/IR/OpenMPDialect.cpp:
```diff
@@ -472,6 +472,99 @@ static void printOrderClause(OpAsmPrinter &p, Operation *op,
+  if (succeeded(parser.parseOptionalKeyword(&enumStr))) {
+    if (std::optional<ClauseType> enumValue = symbolizeClause(enumStr)) {
+      prescriptiveness = ClauseTypeAttr::get(parser.getContext(), *enumValue);
+      if (parser.parseColon())
```
Not a blocking request, but have you considering using a comma instead here? In my opinion, that would look cleaner:
```mlir
omp.taskloop grainsize(strict, %xyz : i64) { ... }
```
Otherwise, using the same character to separate the modifier from the value and the value from the type seems a bit odd to me.

kaviya2510 replied:
Thanks for the review comments @skatrak.
I agree with you. I updated the changes in the recent commit. Kindly go through it.

skatrak replied:
Thank you, looks good!

skatrak reviewed (approved):
LGTM as well, just a small comment on the representation.

kaviya2510 committed:
Merge branch 'main' into grainsize_modifier

kaviya2510 committed:
[MLIR][OpenMP]Replaced colon with a comma separator between strict modifier and operand
